bigquery/tests/system.py

```diff
@@ -1609,6 +1609,42 @@ def test_nested_table_to_dataframe(self):
             row['record_col']['nested_record']['nested_nested_string'],
             'some deep insight')
 
+    def test_list_rows_page_size(self):
+        from google.cloud.bigquery.job import SourceFormat
+        from google.cloud.bigquery.job import WriteDisposition
+        
+        num_items = 7
+        page_size = 3
+        num_pages, num_last_page = divmod(num_items, page_size)
+
+        SF = bigquery.SchemaField
+        schema = [SF('string_col', 'STRING', mode='NULLABLE')]
+        to_insert = [{'string_col': 'item%d' % i} for i in range(num_items)]
+        rows = [json.dumps(row) for row in to_insert]
+        body = six.StringIO('{}\n'.format('\n'.join(rows)))
```

Comment:
Please use BytesIO instead. StringIO will fail if given non-ASCII characters, so I want to make sure we are showing a good example in case someone stumbles onto this.

Reply:
should the other test be changed also?

Reply:
Yes, that would be helpful. If you feel like doing that, let's do it as a separate clean-up PR.

Reply:
might as well since I'm here already.